### PR TITLE
Update LLDP discovery to use IP address if discovery_by_ip is enabled

### PR DESCRIPTION
**Pull request by geordish**
DO NOT DELETE THIS TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [x] Have you signed the [Contributors agreement](http://docs.librenms.org/General/Contributing/) - please do NOT submit a pull request unless you have (signing the agreement in the same pull request is fine). Your commit message for signing the agreement must appear as per the docs.
- [x] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`

Fix for #5872

### CODE REVIEW

File: includes/discovery/discovery-protocols.inc.php [hunk 1]
```diff
@@ -157,6 +157,10 @@
     d_echo($lldp_array);
     $dot1d_array = snmpwalk_cache_oid($device, 'dot1dBasePortIfIndex', array(), 'BRIDGE-MIB');
     d_echo($dot1d_array);
+    if ($config['discovery_by_ip'] == true) {
+        $ptopo_array = snmpwalk_cache_oid($device, 'ptopoConnEntry', array(), 'PTOPO-MIB');
```
Comment (laf) [inline includes/discovery/discovery-protocols.inc.php]: May as well move these two lines into the next `if ($config['discovery_by_ip'] == true) {`

File: includes/discovery/discovery-protocols.inc.php [hunk 2]
```diff
@@ -186,7 +190,24 @@
...
+                                        $a = hexdec($ip_arr[0]);
+                                        $b = hexdec($ip_arr[1]);
+                                        $c = hexdec($ip_arr[2]);
+                                        $d = hexdec($ip_arr[3]);
+                                        
+                                        $discover_hostname = "$a.$b.$c.$d";
+                                    }
+                                 }
+                            } else {
+                                $discover_hostname = $lldp['lldpRemSysName'];
```
Comment (laf) [inline includes/discovery/discovery-protocols.inc.php]: Is lldpRemSysName always a hostname? If it's not then we should try that along with the new code.

Comment (geordish) [inline includes/discovery/discovery-protocols.inc.php]: You mean if discover by IP is enabled, do discover on lldpRemSysName, then discover on IP, if its not, just do discover on lldpRemSysName?

Comment (laf) [inline includes/discovery/discovery-protocols.inc.php]: Well we possibly should actually just run them all regardless, discover_new_device() does checks on discovery_by_ip.

**Comment by towster** [comment]
tested it - detects those devices properly now

**Comment by laf** [comment]
I actually more meant like this. Patch against your branch:

https://gist.githubusercontent.com/laf/3fe8ab5084d896d4cb260124b234bede/raw/00419fbf674a330c4af30a555a7c02c790db5a21/gistfile1.txt

**Comment by geordish** [comment]
Bizarly your patch wouldn't apply!

This should be roughly what you are suggesting. @towster Any chance of testing this updated patch?

**Comment by towster** [comment]
[discovery-10.110.200.60.txt](https://github.com/librenms/librenms/files/791096/discovery-10.110.200.60.txt)

Here is rediscover with patch applied.
It did not add the switch properly..

**Comment by laf** [comment]
I can't see why initially but the logs indicate a bad hostname/ip is being passed. It definitely gets the right part of the code though. @geordish feel free to revert to your known working one.

**Comment by geordish** [comment]
I've added more debugging. @towster - mind providing the output now?

**Comment by towster** [comment]
I re-ran it. Think key is there appears to be a trailing space..
`Testing 'A8 97 DC 98 D2 00' == 'A8 97 DC 98 D2 00 '
`

[discovery-10.110.200.60.txt](https://github.com/librenms/librenms/files/794149/discovery-10.110.200.60.txt)

**Comment by geordish** [comment]
Good spot @towster. That should be fixed now!

**Comment by towster** [comment]
tested it - switch was added properly
Thanks @geordish

**Comment by towster** [comment]
capture

**Comment by geordish** [comment]
was that working before the patch?

**Comment by towster** [comment]
dont believe so - either way the discovery is adding the device properly - to see about resolving the fact that one port links but the other doesnt I would think another issue might be in order - lets see what @laf says

**Comment by laf** [comment]
A debug at the time of it adding will be needed to see what's next but some port info is built up from mac addresses then a discover_link is run.

@geordish If you can remove the d_echo("Testing line then I'm ok to merge this unless you want to look at the link issue as well.

**Comment by geordish** [comment]
Since the issue isn't related to this patch, I will leave it. Don't fancy heading down that rabbit hole right now!

Removed excess debug statements.

**Pull request merged**